Fix model targets and add additional improvements

n1hility opened the pull request:
- Fix model download container and targets
- Add prometheus model for eval
- Improve caching in instructlab container
- Add additional "models" targets for all permutations (cloud-init & bare metal)
- Introduce build chaining so that you can build everthing in one step
- Small update to conform to $(MAKE) convention for submakes

n1hility commented:
I'd like to do a little more testing on this before merging. 

PTAL @rhatdan @Gregory-Pereira 

Note this includes a design change, which might provoke some debate:  build targets propagate to their deps, so you no longer need to build individual elements separately. The UX was a bit too cumbersome without this. For example to build a cloud init image with models you would have needed to do:

make instructlab-gpu
make gpu
make cloud-gpu
make bootc-models MODEL_FROM=<cloud-gpu>

After this change you just do make cloud-gpu-models

rhatdan commented:
These LGTM, but I think to make Konflux team happy we need to move to a multi-stage build for this rather then relying on Makefiles, if possible.

rhatdan commented:
Needs a rebase.

Gregory-Pereira commented:
> These LGTM, but I think to make Konflux team happy we need to move to a multi-stage build for this rather then relying on Makefiles, if possible.

+1 to what Dan said, I was under the assumption we would need to re-write the makefile functionality when moving these into Konflux. Also want to confirm that we are or are not shipping the cloud-init images? I never ended up fixing them, so if this is a priority I can invest some cycles. Aside from rebase + comments mentioned /lgtm

javipolo commented:
We should either merge this ASAP or else we should merge a smaller PR with only the fixes to the model targets, that are currently broken

I prepared https://github.com/containers/ai-lab-recipes/pull/631 with the fixes that would make the pipeline work again, in case we want to continue improving this actual PR

n1hility commented:
sorry for the delay was held up in meetings will rebase it now.

n1hility commented:
There's one other issue introduced by another PR causing an issue fixing sorry for delay

n1hility committed:
Fix model targets and add additional improvements

- Fix model download container and targets
- Add prometheus model for eval
- Improve caching in instructlab container
- Add additional "models" targets for all permutations
- Introduce build chaining so that you can build everthing in one step
- Small update to conform to $(MAKE) convention for submakes

Signed-off-by: Jason T. Greene <jason.greene@redhat.com>

n1hility commented:
PTAL @rhatdan @Gregory-Pereira 

 I think we can probably do more with Multistage in the near future

rhatdan commented:
Broken again.  Needs a rebase.

n1hility commented:
I guess I should have put a hold on the dup PR

Gregory-Pereira reviewed (approved):
/lgtm
